src/main/java/gui/view/MainPageController.java

```diff
@@ -0,0 +1,159 @@
+import gui.viewmodel.ViewModel;
+import javafx.fxml.FXML;
+import javafx.fxml.Initializable;
+import javafx.scene.control.Label;
+import javafx.scene.layout.HBox;
+import javafx.scene.text.Font;
+
+import java.net.URL;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.FormatStyle;
+import java.util.*;
+
+public class MainPageController implements Initializable, ViewModelBindingController {
+
+    private MainPageViewModel viewModel;
+    final double labelFontSize = 15;
+
+    @FXML
+    private JFXDrawer collapsedNavPanel;
+
+    @FXML
+    private JFXDrawer extendedNavPanel;
+
+    @FXML
+    private JFXListView<HBox> eventListView;
+
+    @FXML
+    private JFXListView<HBox> taskListView;
+
+
+    @Override
+    public void initialize(URL location, ResourceBundle resources) {
+        NavigationHelper.initializeNavPanel(extendedNavPanel, collapsedNavPanel);
+    }
+
+    @Override
+    public void init(ViewModel viewModel) {
+        this.viewModel = (MainPageViewModel) viewModel;
+
+        this.viewModel.addObserver(this::acceptReaders);
+
+        updateListView(taskListView, formatTaskReaders(this.viewModel.getRelevantTasks()));
+        updateEventListView(eventListView, formatEventReaders(this.viewModel.getRelevantEvents()));
+
+    }
+
+    /**
+     * Note: removes visibility of all previously rendered labels in the list view before flushing it out.
+     * @param listView the list view to update
+     * @param map a mapping of name and time for today's relevant items
+     */
+    private void updateListView(JFXListView<HBox> listView, Map<String, String> map) {
+        listView.getItems().forEach(i -> i.setVisible(false));
+        listView.getItems().clear();
+        for (Map.Entry<String, String> taskInfo : map.entrySet()) {
+            Label taskName = new Label(taskInfo.getKey());
+            Label deadline = new Label(taskInfo.getValue());
+            taskName.setFont(new Font(labelFontSize));
+            deadline.setFont(new Font(labelFontSize));
+            taskName.setMinWidth(550);
+            taskName.setMaxWidth(550);
+            HBox task = new HBox(taskName, deadline);
+            listView.getItems().add(task);
+        }
+        listView.refresh();
+
+    }
+
+    /**
+     * Note: removes visibility of all previously rendered labels in the list view before flushing it out.
+     * @param eventListView the list view to update
+     * @param map a mapping of name and time for today's relevant items
+     */
+    private void updateEventListView(JFXListView<HBox> eventListView, Map<String, List<String>> map) {
+        eventListView.getItems().forEach(i -> i.setVisible(false));
+        eventListView.getItems().clear();
+        for (Map.Entry<String, List<String>> eventInfo : map.entrySet()) {
+            Label eventName = new Label(eventInfo.getKey());
+            Label startTime = new Label("Start Time is " + eventInfo.getValue().get(0));
```

Comment:
Rather than display "Start Time is...     End Time is ....", I think it can just display "`SartTime` - `EndTime`"